lib/jekyll/admin/api_handler.rb

```diff
@@ -0,0 +1,30 @@
+module Jekyll
+  module Admin
+    class ApiHandler
+      def initialize(site)
+        @site = site
+      end
+
+      def read_file(path)
+        File.read(path)
+      end
+
+      def write_file(path, data)
+        File.write(path, data)
+      end
+
+      def delete_file(path)
+        File.delete(path)
+      end
+
+      def parse_frontmatter(url)
```

Comment:
Same here as above. Whenever possible, we want to use Jekyll's representation of the file, rather than reading and parsing it ourselves.